packages/mobile/src/invite/saga.ts

```diff
@@ -43,41 +44,40 @@ import { sendTransaction } from 'src/transactions/send'
 import { dynamicLink } from 'src/utils/dynamicLink'
 import Logger from 'src/utils/Logger'
 import { web3 } from 'src/web3/contracts'
-import { fetchGasPrice } from 'src/web3/gas'
 import { createNewAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
 import { currentAccountSelector } from 'src/web3/selectors'
 
 const TAG = 'invite/saga'
 export const TEMP_PW = 'ce10'
 export const REDEEM_INVITE_TIMEOUT = 1 * 60 * 1000 // 1 minute
-
-const USE_REAL_FEE = false
 const INVITE_FEE = '0.2'
 
-// TODO(Rossy) Cache this so we don't recalculate it every time we invite someone
-// Especially relevant when inviting many friends
-export async function getInvitationVerificationFee() {
-  // TODO(cmcewen): don't use this
-  if (!USE_REAL_FEE) {
-    return new BigNumber(web3.utils.toWei(INVITE_FEE))
-  }
-  const attestationsContract = await getAttestationsContract(web3)
-  const stableTokenContract = await getStableTokenContract(web3)
-  const verificationFee = await getAttestationFee(
-    attestationsContract,
-    stableTokenContract,
-    NUM_ATTESTATIONS_REQUIRED
-  )
-
-  const gasPrice = await fetchGasPrice()
-  // TODO: estimate gas properly
-  const gasFee = new BigNumber(INVITE_REDEMPTION_GAS).times(gasPrice)
-
-  // We multiply by two to provide a buffer in the event that some requests fail.
-  return verificationFee
-    .times(NUM_ATTESTATIONS_REQUIRED)
-    .plus(gasFee)
-    .times(2)
+export async function getInviteTxGas(
+  account: string,
+  contractGetter: typeof getStableTokenContract | typeof getGoldTokenContract,
+  amount: string,
+  comment: string
+) {
+  const escrowContract = await getEscrowContract(web3)
+  return getSendTxGas(account, contractGetter, {
+    amount,
+    comment,
+    recipientAddress: escrowContract._address,
+  })
+}
+
+export async function getInviteFee(
+  account: string,
+  contractGetter: typeof getStableTokenContract | typeof getGoldTokenContract,
+  amount: string,
+  comment: string
+) {
+  const gas = await getInviteTxGas(account, contractGetter, amount, comment)
+  return (await calculateFee(gas)).plus(getInvitationVerificationFee())
+}
+
+export function getInvitationVerificationFee(inWei: boolean = true) {
```

Comment:
I don't love that this is a boolean arg - unclear from reading the caller why it is false/true and what that means

Reply:
Okay will break apart into two util functions for clarity